backend/eventloop/application_event_loop/application_event_runner_test.go

```diff
@@ -160,6 +161,59 @@ var _ = Describe("ApplicationEventLoop Test", func() {
 
 		})
 
+		It("create an invalid deployment and ensure it fails.", func() {
+			ctx := context.Background()
+
+			scheme, argocdNamespace, kubesystemNamespace, workspace, err := eventlooptypes.GenericTestSetup()
+			Expect(err).To(BeNil())
+
+			gitopsDepl := &managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      strings.Repeat("abc", 100),
+					Namespace: workspace.Name,
+					UID:       uuid.NewUUID(),
+				},
+			}
+
+			informer := sharedutil.ListEventReceiver{}
+
+			k8sClientOuter := fake.NewClientBuilder().WithScheme(scheme).WithObjects(gitopsDepl, workspace, argocdNamespace, kubesystemNamespace).Build()
+			k8sClient := &sharedutil.ProxyClient{
```

Comment:
I think you aren't using the ProxyClient/ListenEventReceiver functionality in this test? So this could be removed.